packages/amplify-graphql-model-transformer/src/resolvers/generators/rds-vtl-generator.ts

```diff
@@ -2,50 +2,59 @@ import {
   generateDefaultLambdaResponseMappingTemplate,
   generateGetLambdaResponseTemplate,
   generateLambdaRequestTemplate,
+  generateCreateInitSlotTemplate,
+  generateLambdaCreateRequestTemplate,
+  generateUpdateInitSlotTemplate,
+  generateLambdaUpdateRequestTemplate,
+  generateLambdaDeleteRequestTemplate,
 } from '../rds';
-import { 
-  ModelCreateInitSlotConfig, 
-  ModelCreateRequestConfig, 
-  ModelDefaultResponseConfig, 
-  ModelRequestConfig, 
-  ModelUpdateRequestConfig, 
-  ModelVTLGenerator, 
-} from "./vtl-generator";
+import {
+  generateSubscriptionRequestTemplate,
+  generateSubscriptionResponseTemplate,
+} from '../dynamodb';
+import {
+  ModelCreateInitSlotConfig,
+  ModelCreateRequestConfig,
+  ModelDefaultResponseConfig,
+  ModelRequestConfig,
+  ModelUpdateRequestConfig,
+  ModelVTLGenerator,
+} from './vtl-generator';
 
 // TODO: This class is created only to show the class structure. This needs a revisit to generate correct resolvers for RDS.
 export class RDSModelVTLGenerator implements ModelVTLGenerator {
   generateUpdateRequestTemplate(config: ModelUpdateRequestConfig): string {
-    return generateLambdaRequestTemplate(config.modelName, config.operation, config.operationName);
+    return generateLambdaUpdateRequestTemplate(config.modelName, config.operationName, config.modelIndexFields ?? ['id']);
   }
   generateCreateRequestTemplate(config: ModelCreateRequestConfig): string {
-    return generateLambdaRequestTemplate(config.modelName, config.operation, config.operationName);
+    return generateLambdaCreateRequestTemplate(config.modelName, config.operationName);
   }
   generateCreateInitSlotTemplate(config: ModelCreateInitSlotConfig): string {
-    return generateDefaultLambdaResponseMappingTemplate(false);
+    return generateCreateInitSlotTemplate(config.modelConfig);
   }
   generateDeleteRequestTemplate(config: ModelUpdateRequestConfig): string {
-    return generateLambdaRequestTemplate(config.modelName, config.operation, config.operationName);
+    return generateLambdaDeleteRequestTemplate(config.modelName, config.operationName, config.modelIndexFields ?? ['id']);
```

Comment:
What's the purpose of `modelIndexFields`? Is it meant to retrieve primary key(s)? If so, primaryKey transformer already does this.

Reply:
I'll work on this, I wrote this because I didn't have primary key support for RDS in yet